tests/components/emulated_hue/test_hue_api.py

```diff
@@ -1042,3 +1060,125 @@ async def test_unauthorized_user_blocked(hue_client):
+    # We simulate a long latence for the actual setting of the entity by forcibly sitting different values directly.
+    await hass_hue.services.async_call(
+        light.DOMAIN,
+        const.SERVICE_TURN_ON,
+        {const.ATTR_ENTITY_ID: "light.ceiling_lights", light.ATTR_BRIGHTNESS: 153},
+        blocking=True,
+    )
+
+    # go through api to get the state back, the value returned should match those set in the last PUT request.
+    ceiling_json = await perform_get_light_state(
+        hue_client, "light.ceiling_lights", HTTP_OK
+    )
+
+    assert ceiling_json["state"][HUE_API_STATE_HUE] == 4369
+    assert ceiling_json["state"][HUE_API_STATE_SAT] == 127
+    assert ceiling_json["state"][HUE_API_STATE_BRI] == 254
+
+    # Make sure that the GET response does not use the cache if PUT response within 2 seconds if the service call is Unsuccessful and the state does not change.
+    await hass_hue.services.async_call(
+        light.DOMAIN,
+        const.SERVICE_TURN_OFF,
+        {const.ATTR_ENTITY_ID: "light.ceiling_lights"},
+        blocking=True,
+    )
+
+    # go through api to get the state back
+    ceiling_json = await perform_get_light_state(
+        hue_client, "light.ceiling_lights", HTTP_OK
+    )
+
+    # Now it should be the real value as the state of the entity has changed to OFF.
+    assert ceiling_json["state"][HUE_API_STATE_HUE] == 0
+    assert ceiling_json["state"][HUE_API_STATE_SAT] == 0
+    assert ceiling_json["state"][HUE_API_STATE_BRI] == 1
+
+    # Ensure we read the actual value after exceeding the timeout time.
+
+    # Turn the bedroom light back on first
+    await hass_hue.services.async_call(
+        light.DOMAIN,
+        const.SERVICE_TURN_ON,
+        {const.ATTR_ENTITY_ID: "light.ceiling_lights"},
+        blocking=True,
+    )
+
+    # update light state through api
+    await perform_put_light_state(
+        hass_hue,
+        hue_client,
+        "light.ceiling_lights",
+        True,
+        hue=4369,
+        saturation=127,
+        brightness=254,
+    )
+
+    await hass_hue.services.async_call(
+        light.DOMAIN,
+        const.SERVICE_TURN_ON,
+        {
+            const.ATTR_ENTITY_ID: "light.ceiling_lights",
+            light.ATTR_BRIGHTNESS: 127,
+            light.ATTR_RGB_COLOR: (1, 2, 7),
+        },
+        blocking=True,
+    )
+
+    # go through api to get the state back, the value returned should match those set in the last PUT request.
+    ceiling_json = await perform_get_light_state(
+        hue_client, "light.ceiling_lights", HTTP_OK
+    )
+
+    # With no wait, we must be reading what we set via the PUT call.
+    assert ceiling_json["state"][HUE_API_STATE_HUE] == 4369
+    assert ceiling_json["state"][HUE_API_STATE_SAT] == 127
+    assert ceiling_json["state"][HUE_API_STATE_BRI] == 254
+
+    await asyncio.sleep(STATE_CACHED_TIMEOUT + 0.1)
```

Comment:
```suggestion
    await asyncio.sleep(0.001)
```

With it patched above we can reduce the sleep